### ISSUE TITLE
Inference of parameter fail

### ISSUE THREAD

**Issue by MrDesjardins** [microsoft/TypeScript#28747]
**TypeScript Version:**  3.2.1
**Search Terms:** vscode, typescript, parameter, infer, inference

**Code**
Private repository so I cannot share the repro. However, here is the idea:

Work if one parameter is not defined:
![2018-11-29_13-31-06](https://user-images.githubusercontent.com/1014796/49254344-e039b380-f3dd-11e8-94c6-25ca817feeea.gif)

Does not work if all of them are not defined:
![2018-11-29_18-22-39](https://user-images.githubusercontent.com/1014796/49264666-0d995800-f404-11e8-9286-dae757caad01.gif)

**Expected behavior:**
All the parameter that can be infered to be infered. Does that cannot to be `any`.

**Actual behavior:**
The feature does not appear at all. No clue why is given.

**Playground Link:** Not available.

**Related Issues:** No

**Comment by MrDesjardins** [comment]
The culprit is the: `FillWindowsParentLevel1: React.ComponentClass<FillWindowsProps> | undefined``
It is the only one that individually, the inference feature does not work.

Here is the code (adjusted for this bug report):
``` 
export function OrgFillWindowsTab(
    activeOrgId: number,
    fillWindows: FillWindowClusters,
    fillWindowsDispatch: FillWindowsDispatch,
    FillWindowsParentLevel1, /*Cannot be inferred*/
    history: History
): SemanticTabPane {
    return {
        menuItem: /*Retracted*/,
        render: () => {
            if (FillWindowsParentLevel1 !== undefined) {
                return (
                    <Tab.Pane /*Retracted*/>
                        {withLogRender(
                            <Segment /*Retracted*/>
                                 /*Retracted*/
                                <FillWindowsParentLevel1
                                    /*Retracted*/
                                />
                            </Segment>
                        )}
                    </Tab.Pane>
                );
            } else {
                // ...
            }
        }
    };
}
```

If the inference cannot be computed for 1 parameter, all the other ones should be infered (and to show the light bulb).

**Comment by weswigham** [comment]
@sandersn does infer from usage have the ability to infer from jsx tag usages?

**Comment by sandersn** [comment]
No, but it would probably not be hard, at least for simple usages.

However, even without *any* usages, the refactoring should apply and produce 'any'. It looks like it's not applying at all there.

**Comment by RyanCavanaugh** [comment]
Closing language service bugs related to the 6.0 implementation. For more information, see #62827
